src/repl/scala/tools/nsc/interpreter/ILoop.scala

```diff
@@ -175,10 +175,18 @@ class ILoop(in0: Option[BufferedReader], protected val out: JPrintWriter)
     echo("\n" + msg)
     in.redrawLine()
   }
-  protected def echo(msg: String) = {
+  protected var mum = false
+  protected def echo(msg: String) = if (!mum) {
     out println msg
     out.flush()
   }
+  // turn off intp reporter and our echo
+  def mumly[A](op: =>A): A =
+    if (isReplDebug) op
+    else intp beSilentDuring {
+      mum = true
+      try op finally mum = false
```

Comment:
you could save the current state of `mum` in a local and finally restore its original value, so nested calls of `mumly` calls would also work